infra/growth/src/resources/eventLogger.js

```diff
@@ -0,0 +1,56 @@
+const Web3 = require('web3')
+
+const { GrowthEvent } = require('@origin/growth-shared/src/resources/event')
+const { GrowthEventTypes } = require('@origin/growth-shared/src/enums')
+
+const logger = require('../logger')
+
+/**
+ * Endpoint for logging entries in the growth_event table.
+ * For now we only support logging events of the BrowserExtensionInstalled type.
+ * Expected paramaters:
+ *  - type: "BrowserExtensionInstalled"
+ *  - eth_address: eth address the user used when signing up for origin rewwards.
+ *  - uid: unique browser extension id
+ * @param req
+ * @param res
+ * @returns {Promise<*>}
+ */
+async function logEvent(req, res) {
+  // Check presence and validity of the type parameter.
+  const type = req.query.type
```

Comment:
@franckc Are we going to blindly trust the client? There isn't much we can do but I'm wondering if there is something we can do to make sure that the API call isn't called by the user manually to fool the system?

Reply:
There won't be any login required after installing the browser extension. So we can't authenticate the user in the backend.

We could perhaps come up with some sort of encoding scheme that uses the eth address to generate a token that then gets verified by the backend. For ex: token = hash(secret+eth address), with secret being hardcoded in the browser extension code (in some obfuscated way) and also being known of the server. It's not bullet proof since a  savvy user could extract the secret from the extension source code.

Now, the payout for the reward not only rely on the presence of this event. It will also be contingent on having recorded browsing activity via the extension throughout the campaign.

So not sure this verification token is necessary. I'd prefer keeping things simple?

Reply:
CC @tomlinton in case he has any thoughts on this topic...

Reply:
Yea I'm not worried about it since they need browsing activity anyway